pkgs/dart_mcp_server/lib/src/mixins/dtd.dart

```diff
@@ -114,12 +117,32 @@ base mixin DartToolingDaemonSupport
               '"${debugSession.name}".',
         );
         addResource(resource, (request) async {
-          return ReadResourceResult(
+          final watch = Stopwatch()..start();
+          final result = ReadResourceResult(
             contents: [
               for (var error in errorService.errorLog.errors)
                 TextResourceContents(uri: resource.uri, text: error),
             ],
           );
+          watch.stop();
+          try {
+            analytics?.send(
+              ua.Event.dartMCPEvent(
+                client: clientInfo.name,
+                clientVersion: clientInfo.version,
+                serverVersion: implementation.version,
+                type: AnalyticsEvent.readResource.name,
+                additionalData: ReadResourceMetrics(
+                  kind: 'runtimeErrors',
```

Comment:
nit: consider keeping constants in the analytics file or in another analytics_constants.dart file so that that it is easy to locate what strings to query for.